frontend/web-2-fac-auth-demo/src/App.js

```diff
@@ -70,14 +75,12 @@ function App() {
             },
           }
         );
-        return firebase.auth().currentUser.getIdToken(true);
       })
-      .then((idToken) => {
-        setLoginToken(idToken);
-        localStorage.setItem("loginToken", idToken);
+      .then(() => {
+        let token = localStorage.getItem("loginToken");
```

Comment:
Are loginTokens session-level or server-level? If it's not session-level, this could be dangerous when dealing w multiple requests.

Reply:
please ignore my stupid comment localStorage is fine to use on the client side LOL